data-es6.js

```diff
@@ -10540,12 +10543,12 @@ exports.tests = [
         set.add(123);
         set.add(123);
 
-        return set.has(123);
+        return set instanceof S && set.has(123);
```

Comment:
What I mean is that prior to ES6, you could not properly subclass builtins - meaning, internal slots (things users can't control) wouldn't be set up. For example, `String.prototype.valueOf.call(foo)` would always throw, even if `foo` was an instance of `class Foo extends String { }` or the equivalent.

The *sole* new behavior that ES6 allows is that using `extends`, and calling `super` in the constructor, sets up the internal slots, so these things don't throw.

`instanceof` merely tells you which `[[Prototype]]` will be referenced - that something has `Set.prototype` in its prototype chain **does not** mean it's a `Set`. The only way you can know it's truly a `Set` is if you use one of the builtin methods that throws when internal slots are missing.

I've had lots of debates about "branding" with the committee, and this is the method they all agreed must be used for branding - and as such, it's the only way to accurately test for builtin-subclassability.